terraform/modules/forest_node/main.tf

```diff
@@ -0,0 +1,90 @@
+terraform {
+  required_version = "~> 1.3"
+
+  required_providers {
+    digitalocean = {
+      source  = "digitalocean/digitalocean"
+      version = "~> 2.0"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = "~> 2.1"
+    }
+  }
+}
+
+provider "digitalocean" {
+  token = var.digitalocean_token
+}
+
+data "digitalocean_ssh_keys" "keys" {
+  sort {
+    key       = "name"
+    direction = "asc"
+  }
+}
+
+data "digitalocean_project" "forest_project" {
+  name = var.project
+}
+
+resource "digitalocean_project_resources" "connect_forest_project" {
+  project   = data.digitalocean_project.forest_project.id
+  resources = [digitalocean_droplet.forest.urn, digitalocean_droplet.forest_observability.urn]
+}
+
+resource "digitalocean_droplet" "forest" {
+  image    = var.image
+  name     = var.name
+  region   = var.region
+  size     = var.size
+  ssh_keys = data.digitalocean_ssh_keys.keys.ssh_keys.*.fingerprint
+
+  lifecycle {
+    create_before_destroy = true
+  }
```

Comment:
Is this beneficial?

Reply:
Yes, I believe it is; if we have to shut down some resources and create new ones for any reason, this will minimize service disruptions. So if this resource creation is not done thoroughly, the deletion of old ones cannot happen